### ISSUE TITLE
error when update packages

### ISSUE THREAD

**Issue by chnlich** [conda/conda#1762]
E:\tools\Anaconda>conda install numpy
Fetching package metadata: ....
Solving package specifications: .
Package plan for installation in environment E:\tools\Anaconda:

The following NEW packages will be INSTALLED:

```
msvc_runtime: 1.0.0-vc9_0   [vc9]
wheel:        0.26.0-py27_1
```

The following packages will be UPDATED:

```
conda:        3.14.1-py27_0 --> 3.18.3-py27_0
conda-env:    2.2.3-py27_0  --> 2.4.4-py27_0
numpy:        1.9.2-py27_0  --> 1.10.1-py27_0
pip:          7.0.3-py27_0  --> 7.1.2-py27_0
pyyaml:       3.11-py27_1   --> 3.11-py27_2
requests:     2.7.0-py27_0  --> 2.8.1-py27_0
setuptools:   17.1.1-py27_0 --> 18.4-py27_0
```

Proceed ([y]/n)? y

Unlinking packages ...
An unexpected error has occurred, please consider sending the            |   0%
following traceback to the conda GitHub issue tracker at:

```
https://github.com/conda/conda/issues
```

Include the output of the command 'conda info' in your report.

Traceback (most recent call last):
  File "E:\tools\Anaconda\Scripts\conda-script.py", line 4, in <module>
    sys.exit(main())
  File "E:\tools\Anaconda\lib\site-packages\conda\cli\main.py", line 201, in main
    args_func(args, p)
  File "E:\tools\Anaconda\lib\site-packages\conda\cli\main.py", line 208, in args_func
    args.func(args, p)
  File "E:\tools\Anaconda\lib\site-packages\conda\cli\common.py", line 612, in inner
    return func(args, parser)
  File "E:\tools\Anaconda\lib\site-packages\conda\cli\main_install.py", line 46, in execute
    install.install(args, parser, 'install')
  File "E:\tools\Anaconda\lib\site-packages\conda\cli\install.py", line 413, in install
    plan.execute_actions(actions, index, verbose=not args.quiet)
  File "E:\tools\Anaconda\lib\site-packages\conda\plan.py", line 527, in execute_actions
    inst.execute_instructions(plan, index, verbose)
  File "E:\tools\Anaconda\lib\site-packages\conda\instructions.py", line 148, in execute_instructions
    cmd(state, arg)
  File "E:\tools\Anaconda\lib\site-packages\conda\instructions.py", line 95, in UNLINK_CMD
    install.unlink(state['prefix'], arg)
  File "E:\tools\Anaconda\lib\site-packages\conda\install.py", line 693, in unlink
    os.unlink(meta_path)
WindowsError: [Error 5] : 'E:\tools\Anaconda\conda-meta\conda-3.14.1-py27_0.json'

E:\tools\Anaconda>conda info
Current conda install:

```
         platform : win-64
    conda version : 3.14.1
```

  conda-build version : 1.14.1
       python version : 2.7.10.final.0
     requests version : 2.7.0
     root environment : E:\tools\Anaconda  (writable)
  default environment : E:\tools\Anaconda
     envs directories : E:\tools\Anaconda\envs
        package cache : E:\tools\Anaconda\pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/win-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/win-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False

E:\tools\Anaconda>

**Comment by portig** [comment]
Here is the screen output from a similar error condition. It includes an earlier attempt at updating my Miniconda installation that failed with SSL verification errors.

D:\Tools\Bat>conda update --all
Fetching package metadata: SSL verification error: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:600)
SSL verification error: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:600)
SSL verification error: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:600)
.SSL verification error: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:600)
...
Solving package specifications: .......................................................................................................
Package plan for installation in environment D:\Tools\Anaconda\3:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    conda-env-2.4.4            |           py34_0          64 KB
    menuinst-1.2.1             |           py34_0          45 KB
    tornado-4.2.1              |           py34_1         552 KB
    anaconda-client-1.1.0      |           py34_0         148 KB
    ipython-4.0.0              |           py34_1        1017 KB
    ------------------------------------------------------------
                                           Total:         1.8 MB

The following packages will be UPDATED:

    anaconda-client: 1.0.2-py34_0 --> 1.1.0-py34_0
    conda-env:       2.4.2-py34_0 --> 2.4.4-py34_0
    ipython:         4.0.0-py34_0 --> 4.0.0-py34_1
    menuinst:        1.1.0-py34_0 --> 1.2.1-py34_0
    tornado:         4.2.1-py34_0 --> 4.2.1-py34_1

Proceed ([y]/n)? y

Could not connect to https://repo.continuum.io/pkgs/free/win-64/menuinst-1.2.1-py34_0.tar.bz2
Error: Connection error: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:600): https://repo.continuum.io/pkgs/free/win-64/menuinst-1.2.1-py34_0.tar.bz2

D:\Tools\Bat>conda config --set ssl_verify false
D:\Tools\Bat>conda update --all
Fetching package metadata: ....
Solving package specifications: .........................................................................................................
Package plan for installation in environment D:\Tools\Anaconda\3:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    conda-env-2.4.4            |           py34_0          64 KB
    tornado-4.2.1              |           py34_1         552 KB
    anaconda-client-1.1.0      |           py34_0         148 KB
    ipython-4.0.0              |           py34_1        1017 KB
    ------------------------------------------------------------
                                           Total:         1.7 MB

The following packages will be UPDATED:

    anaconda-client: 1.0.2-py34_0 --> 1.1.0-py34_0
    conda-env:       2.4.2-py34_0 --> 2.4.4-py34_0
    ipython:         4.0.0-py34_0 --> 4.0.0-py34_1
    tornado:         4.2.1-py34_0 --> 4.2.1-py34_1

Proceed ([y]/n)? y

Fetching packages ...
conda-env-2.4. 100% |###############################| Time: 0:00:00 672.05 kB/s
tornado-4.2.1- 100% |###############################| Time: 0:00:00   2.10 MB/s
anaconda-clien 100% |###############################| Time: 0:00:00   1.10 MB/s
ipython-4.0.0- 100% |###############################| Time: 0:00:00   3.16 MB/s
Extracting packages ...
[      COMPLETE      ]|##################################################| 100%
Unlinking packages ...
WARNING:root:Menuinst could not be imported:####                         |  50%
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda/issues

Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
  File "D:\Tools\Anaconda\3\lib\site-packages\conda\install.py", line 326, in mk_menus import menuinst
ImportError: No module named 'menuinst'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\Tools\Anaconda\3\Scripts\conda-script.py", line 4, in <module> sys.exit(main())
  File "D:\Tools\Anaconda\3\lib\site-packages\conda\cli\main.py", line 195, in main args_func(args, p)
  File "D:\Tools\Anaconda\3\lib\site-packages\conda\cli\main.py", line 202, in args_func args.func(args, p)
  File "D:\Tools\Anaconda\3\lib\site-packages\conda\cli\main_update.py", line 48, in execute install.install(args, parser, 'update')
  File "D:\Tools\Anaconda\3\lib\site-packages\conda\cli\install.py", line 423, in install plan.execute_actions(actions, index, verbose=not args.quiet)
  File "D:\Tools\Anaconda\3\lib\site-packages\conda\plan.py", line 537, in execute_actions inst.execute_instructions(plan, index, verbose)
  File "D:\Tools\Anaconda\3\lib\site-packages\conda\instructions.py", line 148, in execute_instructions cmd(state, arg)
  File "D:\Tools\Anaconda\3\lib\site-packages\conda\instructions.py", line 95, in UNLINK_CMD install.unlink(state['prefix'], arg)
  File "D:\Tools\Anaconda\3\lib\site-packages\conda\install.py", line 692, in unlink mk_menus(prefix, meta['files'], remove=True)
  File "D:\Tools\Anaconda\3\lib\site-packages\conda\install.py", line 329, in mk_menus logging.warn(e.message)
AttributeError: 'ImportError' object has no attribute 'message'
D:\Tools\Bat>conda info
Current conda install:

             platform : win-64
        conda version : 3.18.3
  conda-build version : not installed
       python version : 3.4.3.final.0
     requests version : 2.8.1
     root environment : D:\Tools\Anaconda\3  (writable)
  default environment : D:\Tools\Anaconda\3
     envs directories : D:\Tools\Anaconda\3\envs
        package cache : D:\Tools\Anaconda\3\pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/win-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/win-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : D:\Users\portig\.condarc
    is foreign system : False

D:\Tools\Bat>conda config --set ssl_verify true

**Comment by msarahan** [comment]
.@portig your error is different, but easier. Menuinst somehow didn't get picked up as a dependency the second time you tried to update. 

Please install this yourself with

    Conda install menuinst 

We also need to fix conda so that that import error fails more gracefully. Menuinst is not strictly required. It sets up start menu shortcuts for packages that have them.

**Comment by msarahan** [comment]
@chnlich I think conda is trying to delete a file that is not there.  Did you delete files in the installation yourself?  Again, we can improve error handling in conda so that this condition is not fatal to updates. However, this fix can't be retroactive: if updating is broken in an old version, you can't update to the new version to get the fix. 

Regardless, the simplest fix is to reinstall Miniconda or Anaconda.

**Comment by portig** [comment]
@msarahan Installing menuinst helped me complete the "conda update --all" process. Thanks.

However before fully succeeding I had problems with "Unsatisfiable package specifications". The hint that was generated was that msvc_runtime had unsatisfiable dependencies. I ended up removing msvc_runtime. My main reason for using Miniconda is to be able to use matplotlib in connection with Julia 0.4. This still works satisfactorily.

The only current problem with my Miniconda installation is that I have to turn off ssl_verify because otherwise I get SSL certificate verify failures. This does not seem to be an ideal thing to have to do.

**Comment by msarahan** [comment]
@portig this SSL stuff is a known issue.  We have updated OpenSSL, certifi, and cryptography so that they are current, to maintain the security of our users.  Certifi in particular revoked many certificates, causing the issues you're having.  They did so for the best reason possible: those certificates were no longer secure.  Here is an excerpt from a pending description of the problem and how to fix it:

```
Problems arise when certifi is updated prior to updating openssl and cryptography.  If you are seeing SSL errors, the best way to restore your system is to:

Set ssl_verify to False in .condarc
In .condarc, set the following
ssl_verify: False
Note that this DISABLES verification of SSL certificiates.
Upgrade openssl and cryptography:
conda update openssl cryptography
Update certifi:
conda update certifi

Re-enable SSL verification by removing the ssl_verify line from .condarc
```

I hope this helps get you back on track.

**Comment by chnlich** [comment]
@msarahan Thank you, though I don't know why it doesn't work, I try to reinstall the anaconda and it works well.

**Comment by portig** [comment]
@msarahan I followed your recommendation to update openssl, cryptography and certifi. They all had to be installed first because they did not exist yet on my machine. However, after resetting ssl_verify to true the SSL errors still pop up and there is a report about an unsatisfiable package specification. Here is the session log:

D:\Users\portig>conda config --set ssl_verify true
D:\Users\portig>conda update --all
Fetching package metadata: SSL verification error: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:600)
SSL verification error: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:600)
SSL verification error: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:600)
SSL verification error: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:600)
....
Solving package specifications: .......................................................................................
Error: Unsatisfiable package specifications.
Generating hint:
[      COMPLETE      ]|##################################################| 100%


Hint: 'openssl' has unsatisfiable dependencies (see 'conda info openssl')

Note that the following features are enabled:
  - vc9
  - vc14
  - vc10

**Comment by chadkennedyonline** [comment]
Looks like the same issue here: #1779

**Comment by ijstokes** [comment]
@msarahan of course we need to appreciate that turning OFF SSL verification immediately before installing `openssl` and `certifi` packages is the worse possible set of packages for which we could do this.  But I appreciate that it is a chicken and egg problem.  Nonetheless, if there is any *better* way for users to resolve this issue (even if it is only an *alternative* resolution) in a way that has some semblance of security then we should definitely present that strategy.

**Comment by kalefranz** [comment]
Original stack trace fixed with conda 4.3.
